src/esp/sensor/Sensor.h

```diff
@@ -37,14 +37,20 @@ enum class ObservationSpaceType {
   TEXT = 2,
 };
 
+enum class SensorSubType {
+  Pinhole = 0,
```

Comment:
Dumb question, suggesting a possible rename -- what's the more common / clearer term? Perspective or pinhole camera?

Reply:
Probably perspective, but Pinhole also implies infinite depth of field, which is appropriate here, right?

Reply:
Does it? I thought the code gives the user an option to set the far plane to a non-infinite value.

Reply:
Well, that's far-clipping, right?  Not blur.  Everything between near and far plane is in focus, so it's like a pinhole camera.

Reply:
I mean, if this were just a graphics app, then yeah, it would go without saying that it should be perspective.  But i can understand that vision folks would look at this and value the distinction that this is a pinhole camera. Not really sure, to be honest.

Reply:
Ah, I see, I see. Yeah you're right, good point with the DOF.

Reply:
This is to distinguish the projection method, which is perspective VS orthogonal.
I am leaning towards using the "Perspective" here. :)

Reply:
But considering this is actually used under the "CameraSensor" and it is to distinguish different camera sensors, maybe "Pinhole" is reasonable in this case. (Later we may have "Fisheye").

So I do not know. It is up to you.

By the way is there any reason why we do not put it under the "CameraSensor" header file?

Reply:
Sensor spec is used for all sensors, not just CameraSensors (especially when others exist :) ), and i envisioned the SensorSubType field as a "catch-all" descriptor for some defining secondary feature of a sensor.  This is why I felt it would be more appropriate in Sensor.h than in CameraSensor.h.